cpp/src/io/parquet/reader_impl.cpp

```diff
@@ -359,58 +386,65 @@ table_with_metadata reader::impl::read_chunk_internal(bool uses_custom_row_bound
                             : std::nullopt;
     // Only construct `out_metadata` if `_output_metadata` has not been cached.
     if (!_output_metadata) {
-      column_name_info& col_name = out_metadata.schema_info.emplace_back("");
+      column_name_info& col_name = out_metadata.schema_info[i];
       out_columns.emplace_back(make_column(_output_buffers[i], &col_name, metadata, _stream));
     } else {
       out_columns.emplace_back(make_column(_output_buffers[i], nullptr, metadata, _stream));
     }
   }
 
-  // Add empty columns if needed.
-  return finalize_output(out_metadata, out_columns);
+  // Add empty columns if needed. Filter output columns based on filter.
+  return finalize_output(out_metadata, out_columns, filter);
```

Comment:
So this reads every column and culls at the end?

Reply:
yes.

Reply:
@hyperbolic2346 since chunked reader does not support user specifying the row group too, filtering is not supported by chunked row group reader.

Reply:
no error is thrown if user specifies the row groups for chunked reader object creation. it's silently ignored. So, the same convention is followed here. @hyperbolic2346 is this behaviour okay?

Reply:
The chunked reader and row-group specification should work together fine (though I don't think we have a test specifically for it).  The only input to the chunked reader is "number of bytes in a chunk" so it doesn't care at all if the row groups being inspected in the first place are the entire file or part of it.